sdk/identity/identity/src/credentials/managedIdentityCredential/arcMsi.ts

```diff
@@ -96,7 +96,19 @@ async function filePathRequest(
  * Defines how to determine whether the Azure Arc MSI is available, and also how to retrieve a token from the Azure Arc MSI.
  */
 export const arcMsi: MSI = {
-  async isAvailable(scopes): Promise<boolean> {
+  async isAvailable({ scopes, clientId, resourceId }): Promise<boolean> {
+    if (clientId) {
+      logger.info(
+        `${msiName}: Unavailable. User assigned identity is not supported by the Azure Arc Managed Identity Endpoint. To authenticate with the system assigned identity, omit the client id when constructing the ManagedIdentityCredential, or if authenticating with the DefaultAzureCredential ensure the AZURE_CLIENT_ID environment variable is not set.`
+      );
+      return false;
```

Comment:
Was it decided that this isn't a breaking change to make an MSI provider fail if `clientId` is provided? For `resourceId` it's fine because it's a new option, but clientId isn't new if someone passed it in the ManagedIdentityCredential constructor.

Reply:
We already used to throw if clientId was provided to this MSI. This is more of a cleanup: Disabling the MSI instead of throwing.

Reply:
Ah I see, the error was thrown in getToken.